cmd/notation/internal/truststore/truststore.go

```diff
@@ -0,0 +1,226 @@
+	}
+
+	// check if the target path is a cert (support PEM and DER formats)
+	if _, err := corex509.ReadCertificateFile(certPath); err != nil {
+		return err
+	}
+
+	// core process
+	// get the trust store path
+	trustStorePath, err := dir.Path.UserConfigFS.GetPath(dir.TrustStoreDir, "x509", storeType, namedStore)
+	if err := CheckNonErrNotExistError(err); err != nil {
+		return err
+	}
+	// check if certificate already in the trust store
+	if _, err := os.Stat(filepath.Join(trustStorePath, filepath.Base(certPath))); err == nil {
+		return errors.New("certificate already exists in the Trust Store")
+	}
+	// add cert to trust store
+	_, err = osutil.CopyToDir(certPath, trustStorePath)
+	if err != nil {
+		return err
+	}
+
+	// write out
+	if display {
+		fmt.Printf("Successfully added %s to named store %s of type %s\n", filepath.Base(certPath), namedStore, storeType)
+	}
+
+	return nil
+}
+
+// ListCerts walks through root and lists all x509 certificates in it,
+// sub-dirs are ignored.
+func ListCerts(root string, depth int) error {
+	maxDepth := strings.Count(root, string(os.PathSeparator)) + depth
+
+	return filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() && strings.Count(path, string(os.PathSeparator)) > maxDepth {
+			return fs.SkipDir
+		}
+		info, err := d.Info()
+		if err != nil {
+			return err
+		}
+		if info.Mode().IsRegular() {
+			if _, err := corex509.ReadCertificateFile(path); err != nil {
+				return err
+			}
+			fmt.Println(path)
+		}
+		return nil
+	})
+}
+
+// ShowCerts writes out details of certificates
+func ShowCerts(certs []*x509.Certificate) {
+	fmt.Println("Display certificate details. Starting from leaf certificate if it's a certificate chain.")
+	fmt.Println("--------------------------------------------------------------------------------")
+	for ind, cert := range certs {
+		showCert(cert)
+		if ind != len(certs)-1 {
+			fmt.Println("--------------------------------------------------------------------------------")
+		}
+	}
+}
+
+// showCert displays details of a certificate
+func showCert(cert *x509.Certificate) {
+	fmt.Println("Issuer:", cert.Issuer)
+	fmt.Println("Subject:", cert.Subject)
+	fmt.Println("Valid from:", cert.NotBefore)
+	fmt.Println("Valid to:", cert.NotAfter)
+	fmt.Println("IsCA:", cert.IsCA)
+
+	h := sha1.Sum(cert.Raw)
+	fmt.Println("Thumbprints:", hex.EncodeToString(h[:]))
+}
+
+// DeleteAllCerts deletes all certificate files from the trust store
+// under dir truststore/x509/storeType/namedStore
+func DeleteAllCerts(storeType, namedStore string, confirmed bool, errorSlice []error) []error {
+	path, err := dir.Path.UserConfigFS.GetPath(dir.TrustStoreDir, "x509", storeType, namedStore)
+	if err == nil {
+		truststorePath := fmt.Sprintf("truststore/x509/%s/%s", storeType, namedStore)
+		prompt := fmt.Sprintf("Are you sure you want to remove all certificate files under: %q?", truststorePath)
```

Comment:
Also since we are managing trust-store can we obfuscate the trust store location from user? And show trust store location only when there is an error.
```suggestion
		prompt := fmt.Sprintf("Are you sure you want to remove all certificate for %s of type %s?", namedStore, storeType)
```

Reply:
Sure.